assert/assertions_test.go

```diff
@@ -3192,3 +3192,45 @@ func TestErrorAs(t *testing.T) {
 		})
 	}
 }
+
+func Test_indentMessageLines(t *testing.T) {
```

Comment:
Move this new test to a separate file `assertions_priv_test.go`: we aim to move `assertions_test.go` to package `assert_test` and this new test will get in the way for that project.

Reply:
> we aim to move assertions_test.go to package assert_test

Is this documented/discussed anywhere? Would love to get more context.

Reply:
Made that change

Reply:
@arjunmahishi No, I've not discussed this anywhere I remember. However, to ease maintenance I think it would be better to have clear separation of unit tests written to check implementation details from higher level unit tests that test the Testify external API.

A related effort is #1431 by @brackendawson. While I approve the principle of that effort, I think we must first clear most of the PR queue to avoid missing a valuable pending bug fix that would be broken by such a major refactor. I have started co-maintenance in July 2023, but so far I haven't been able to review _all_ issues and PRs (those without labels haven't been reviewed).